### PR TITLE
avian: remove

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change
ZHF #122042

Doesn't compile anymore[1] and isn't supported by upstream[2].

[1] https://hydra.nixos.org/build/142613769
[2] https://github.com/ReadyTalk/avian/blob/0871979b298add320ca63f65060acb7532c8a0dd/README.md

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
if it's been abandoned, then we should probably remove it, and add the removal error to `aliases.nix`

**Comment by Ma27** [comment]
@jonringer you're absolutely right! Package will be removed in here now.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -971,6 +971,12 @@ mapAliases ({
 
   posix_man_pages = man-pages-posix; # Added 2021-04-15
 
+  /* Cleanup before 21.11 */
+  avian = throw ''
+    The package doesn't compile anymore on NixOS and both development
+    & maintenance is abandoned by upstream.
+  '';
```
Comment (jonringer) [inline pkgs/top-level/aliases.nix:978]: do you mind adding the `# Added 2021-05-07`.

May help in the future to garbage-collect entries no one cares about in the future

Comment (Ma27) [reply pkgs/top-level/aliases.nix:978]: @jonringer done.

**Review by jonringer — APPROVED** [review]
LGTM

**Pull request merged**